src/game/g_active.c

```diff
@@ -517,38 +535,71 @@ void G_TouchTriggers(gentity_t *ent)
 qboolean G_SpectatorAttackFollow(gentity_t *ent)
 {
 	trace_t       tr;
-	vec3_t        forward, right, up;
-	vec3_t        start, end;
+	vec3_t        forward;
 	vec3_t        mins, maxs;
-	static vec3_t enlargeMins = { -64.0f, -64.0f, -48.0f };
-	static vec3_t enlargeMaxs = { 64.0f, 64.0f, 0.0f };
+	static vec3_t enlargeSmallMins = { -4.0f, -4.0f, -3.0f }, enlargeSmallMaxs = { 4.0f, 4.0f, 0.0f };
+	static vec3_t enlargeMediumMins = { -16.0f, -16.0f, -12.0f }, enlargeMediumMaxs = { 16.0f, 16.0f, 0.0f };
+	static vec3_t enlargeLargeMins = { -64.0f, -64.0f, -48.0f }, enlargeLargeMaxs = { 64.0f, 64.0f, 0.0f };
 
 	if (!ent->client)
 	{
 		return qfalse;
 	}
 
-	AngleVectors(ent->client->ps.viewangles, forward, right, up);
-	VectorCopy(ent->client->ps.origin, start);
-	VectorMA(start, MAX_TRACE, forward, end);
+	AngleVectors(ent->client->ps.viewangles, forward, NULL, NULL);
 
-	// enlarge the hitboxes, so spectators can easily click on them..
-	VectorCopy(ent->r.mins, mins);
-	VectorCopy(ent->r.maxs, maxs);
-	VectorAdd(mins, enlargeMins, mins);
-	VectorAdd(maxs, enlargeMaxs, maxs);
-	// also put the start-point a bit forward, so we don't start the trace in solid..
-	VectorMA(start, 75.0f, forward, start);
+	// let's do 4 traces with increasing "beam width" to prioritize the player
+	// we are directly looking at, but also allow some more fuzzy angles with
+	// decreasing priority
+	//
+	{ // 1st. dead on
+		if (G_SpectatorAttackFollowAttempt(ent, &tr, forward, NULL, NULL))
+		{
+			goto done;
+		}
+	}
+	{ // 2nd. small enlargement
+		VectorCopy(ent->r.mins, mins);
+		VectorCopy(ent->r.maxs, maxs);
+		VectorAdd(mins, enlargeSmallMins, mins);
+		VectorAdd(maxs, enlargeSmallMaxs, maxs);
 
-	// trap_Trace(&tr, start, mins, maxs, end, ent->client->ps.clientNum, CONTENTS_BODY | CONTENTS_CORPSE);
-	G_HistoricalTrace(ent, &tr, start, mins, maxs, end, ent->s.number, CONTENTS_BODY | CONTENTS_CORPSE);
+		if (G_SpectatorAttackFollowAttempt(ent, &tr, forward, mins, maxs))
+		{
+			goto done;
+		}
+	}
+	{ // 3rd. medium enlargement
+		VectorCopy(ent->r.mins, mins);
+		VectorCopy(ent->r.maxs, maxs);
+		VectorAdd(mins, enlargeMediumMins, mins);
+		VectorAdd(maxs, enlargeMediumMaxs, maxs);
 
+		if (G_SpectatorAttackFollowAttempt(ent, &tr, forward, mins, maxs))
+		{
+			goto done;
+		}
+	}
+	{ // 4th. large enlargement
+		VectorCopy(ent->r.mins, mins);
+		VectorCopy(ent->r.maxs, maxs);
+		VectorAdd(mins, enlargeLargeMins, mins);
+		VectorAdd(maxs, enlargeLargeMaxs, maxs);
+
+		if (G_SpectatorAttackFollowAttempt(ent, &tr, forward, mins, maxs))
+		{
+			goto done;
+		}
+	}
```

Comment:
Use a loop and multiply by 4 the box bounds at each iteration.
The first iteration could start with BBox of 1, so it is icncluded in the loop
or
the first trace could start outside the loop, so it keep the non usage of BBox (less computation for the first trace)

Reply:
Done! Ready for re-review :metal: